openedx_events/event_bus/avro/tests/test_avro.py

```diff
@@ -11,10 +18,61 @@
     deserialize_bytes_to_event_data,
     serialize_event_data_to_bytes,
 )
+from openedx_events.learning import signals  # pylint: disable=reimported
+from openedx_events.tooling import OpenEdxPublicSignal
+
+
+def generate_test_event_data_for_signal(class_type):
+    """
+    Generates test data automatically by filling in dummy data for basic data types (int/bool/str)
+    Recursively breaks down the classes for nested classes
+
+    Arguments:
+        class_type: The type of the class which we are generating data for
+
+    Returns:
+        dtdict: A data dictionary containing dummy data for all attributes of the class
+    """
+    dtdict = {}
+    switcher = {
```

Comment:
Nit: a more descriptive variable name would be something like `defaults_per_type`